### PR TITLE
ci: Test with Go 1.19, 1.20

### LINKED ISSUE #12040: Go 1.20 Upgrade
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

This is just a tracking issue for upgrading pu/pu to Go 1.20.
We should link any related tickets here.

The upgrade will help:

- #9224 cc @AaronFriel 

The upgrade is blocked on:

- [x] golangci-lint support with 1.20

Upgrade PR: https://github.com/pulumi/pulumi/pull/12039

### PR DESCRIPTION
**Pull request by abhinav**
Go 1.20 was just released.
Switch to testing with Go 1.19 and 1.20.

Resolves #12040

### CODE REVIEW

**Comment by abhinav** [comment]
Looks like golanglint-ci needs an update.
Right now, it will take up all CPU and memory and not produce anything.

**Comment by abhinav** [comment]
> Looks like golanglint-ci needs an update. Right now, it will take up all CPU and memory and not produce anything.

Compiling it from source with Go 1.20 fixed the issue.
So we just need to wait for a 1.20 build.

**Comment by abhinav** [comment]
Lint fails but I want to try the rest of the tests anyway.

**Comment by abhinav** [comment]
bors try

**Comment by abhinav** [comment]
Blocked on toolchain updates; we'll likely need to wait for a new golanglint-ci release.

File: .github/workflows/ci-test-docs-generation.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.19.x # TODO(friel) use version_set
+          go-version: 1.20.x # TODO(friel) use version_set
```
Comment (RobbieMcKinstry) [inline .github/workflows/ci-test-docs-generation.yml:40]: Any chance we could address this TODO since we're touching this line and not much else? WDYT?

Comment (abhinav) [reply .github/workflows/ci-test-docs-generation.yml:40]: Will explore

**Comment by abhinav** [comment]
Huh. Guess the auto rebase doesn't work.

File: .github/workflows/ci-test-docs-generation.yml (hunk 1 above)
Comment (RobbieMcKinstry) [reply .github/workflows/ci-test-docs-generation.yml:40]: TY!

**Comment by abhinav** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12039** <a href="https://app.graphite.dev/github/pr/pulumi/pulumi/12039" target="_blank">Graphite</a>  👈
    * **PR #12048** <a href="https://app.graphite.dev/github/pr/pulumi/pulumi/12048" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/pulumi/pulumi/12039?utm_source=stack-comment).

File: .github/workflows/ci-test-docs-generation.yml (hunk 1 above)
Comment (abhinav) [reply .github/workflows/ci-test-docs-generation.yml:40]: Trying out in https://github.com/pulumi/pulumi/pull/12048. Not ready for review yet.

**Comment by abhinav** [comment]
bors try

**Comment by abhinav** [comment]
bors try

**Comment by abhinav** [comment]
> Integration tests are timing out on macOS. Will dig in locally.

Actually looks like these are timing out for all PRs

**Comment by abhinav** [comment]
bors try

**Comment by abhinav** [comment]
The 'bors try' is green. Everything passes with 1.20.
Ready for review again.

**Review by justinvp — APPROVED** [review]

**Pull request merged**